File: app/src/main/java/com/example/recipefinalproject/adapters/CategoryAdapter.java
package com.example.recipefinalproject.adapters;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.recipefinalproject.R;
import com.example.recipefinalproject.databinding.ItemCategoryBinding;
import com.example.recipefinalproject.models.Category;

import java.util.List;

public class CategoryAdapter extends RecyclerView.Adapter<CategoryAdapter.CategoryHolder> {
    List<Category> categoryList;

    public void setCategoryList(List<Category> categoryList) {
        this.categoryList = categoryList;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public CategoryAdapter.CategoryHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new CategoryHolder(ItemCategoryBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull CategoryAdapter.CategoryHolder holder, int position) {
        Category category = categoryList.get(position);
        holder.onBind(category);

    }

    @Override
    public int getItemCount() {
        return categoryList.size();
    }

    public static class CategoryHolder extends RecyclerView.ViewHolder {
        private ItemCategoryBinding binding;
        public CategoryHolder(@NonNull ItemCategoryBinding itemView) {
            super(itemView.getRoot());
            binding = itemView;
        }

        public void onBind(Category category) {
            binding.tvName.setText(category.getName());
            if(category.getName().equalsIgnoreCase("Lunch")) {
                binding.imgBgCategory.setImageResource(R.drawable.lunch);
            } else if(category.getName().equalsIgnoreCase("Dinner")) {
                binding.imgBgCategory.setImageResource(R.drawable.recipe1);
            } else {
                binding.imgBgCategory.setImageResource(R.drawable.breakfast);
            }

        }
    }
}

File: app/src/main/java/com/example/recipefinalproject/adapters/CategoryAdapter.kt
package com.example.recipefinalproject.adapters

import android.content.Intent
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.example.recipefinalproject.AllRecipesActivity
import com.example.recipefinalproject.R
import com.example.recipefinalproject.databinding.ItemCategoryBinding
import com.example.recipefinalproject.models.Category

class CategoryAdapter : RecyclerView.Adapter<CategoryAdapter.CategoryHolder>() {
    var categoryList1: List<Category> = ArrayList()

    fun setCategoryList(categoryList: MutableList<Category?>) {
        this.categoryList1 = categoryList.filterNotNull()
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CategoryHolder {
        return CategoryHolder(
            ItemCategoryBinding.inflate(
                LayoutInflater.from(parent.context),
                parent,
                false
            )
        )
    }

    override fun onBindViewHolder(holder: CategoryHolder, position: Int) {
        val category = categoryList1[position]
        holder.onBind(category)
    }

    override fun getItemCount(): Int {
        return categoryList1.size
    }

    class CategoryHolder(private val binding: ItemCategoryBinding) :
        RecyclerView.ViewHolder(binding.root) {
        fun onBind(category: Category) {
            binding.tvName.text = category.name
            Glide.with(binding.root.context)
                .load(category.image)
                .centerCrop()
                .placeholder(R.mipmap.ic_launcher)
                .into(binding.imgBgCategory)

            binding.root.setOnClickListener {
                val intent = Intent(binding.root.context, AllRecipesActivity::class.java)
                intent.putExtra("type", "category")
                intent.putExtra("category", category.name)
                binding.root.context.startActivity(intent)
            }
        }
    }
}